setup.cfg

```diff
@@ -18,6 +18,8 @@ classifier =
 
 [files]
 packages =
+    nova
```

Comment:
Running bdist_wheel with python2.7, the `__init__.py` files in the `nova` and `nova/virt` packages weren't being included.

Reply:
Since the nova-lxd egg installs into the nova egg, should nova-lxd be creating __init__.py files in nova/ and nova/virt/? I would think the ones already there from nova would be the correct ones.

Reply:
A couple of things:

1. A wheel is being created instead of an egg. Apparently, `pip` doesn't support installing eggs, and wheels are intended to replace eggs according to https://pythonwheels.com/.
2. As described in the comment above, running `bdist_wheel`, the `__init__.py` files in the `nova` and `nova/virt` packages weren't included in the packaged wheel, so I had to work around that. I'm open to other suggestions though.

Reply:
Based on our discussion last week, rather than letting `pip` install to the default directory, `/usr/local/lib/python2.7/dist-packages`, I specified the directory `/usr/lib/python2.7/dist-packages`, and because pip detected packages already installed at that location (i.e. /usr/lib/python2.7/dist-packages/nova), I had to specify the `--upgrade` flag. 

This however did not work as desired. These other packages were all removed:

nova/api
nova/api/ec2
nova/api/validation
nova/api/metadata
nova/api/openstack
nova/api/openstack/compute
nova/api/openstack/compute/schemas
nova/api/openstack/compute/views
nova/api/openstack/placement
nova/api/openstack/placement/schemas
nova/api/openstack/placement/handlers
nova/network
nova/network/security_group
nova/network/neutronv2
nova/hacking
nova/keymgr
nova/db
nova/db/sqlalchemy
nova/db/sqlalchemy/migrate_repo
nova/db/sqlalchemy/migrate_repo/versions
nova/db/sqlalchemy/api_migrations
nova/db/sqlalchemy/api_migrations/migrate_repo
nova/db/sqlalchemy/api_migrations/migrate_repo/versions
nova/pci
nova/notifications
nova/notifications/objects
nova/compute
nova/compute/monitors
nova/compute/monitors/cpu
nova/objects
nova/privsep
nova/servicegroup
nova/servicegroup/drivers
nova/policies
nova/image
nova/image/download
nova/vnc
nova/common
nova/cells
nova/cells/filters
nova/cells/weights
nova/consoleauth
nova/locale
nova/locale/cs
nova/locale/cs/LC_MESSAGES
nova/locale/tr_TR
nova/locale/tr_TR/LC_MESSAGES
nova/locale/es
nova/locale/es/LC_MESSAGES
nova/locale/zh_TW
nova/locale/zh_TW/LC_MESSAGES
nova/locale/zh_CN
nova/locale/zh_CN/LC_MESSAGES
nova/locale/fr
nova/locale/fr/LC_MESSAGES
nova/locale/ja
nova/locale/ja/LC_MESSAGES
nova/locale/it
nova/locale/it/LC_MESSAGES
nova/locale/ko_KR
nova/locale/ko_KR/LC_MESSAGES
nova/locale/pt_BR
nova/locale/pt_BR/LC_MESSAGES
nova/locale/de
nova/locale/de/LC_MESSAGES
nova/locale/ru
nova/locale/ru/LC_MESSAGES
nova/volume
nova/cmd
nova/scheduler
nova/scheduler/client
nova/scheduler/filters
nova/scheduler/weights
nova/tests
nova/tests/functional
nova/tests/functional/wsgi
nova/tests/functional/api
nova/tests/functional/api/openstack
nova/tests/functional/api/openstack/placement
nova/tests/functional/api/openstack/placement/gabbits
nova/tests/functional/regressions
nova/tests/functional/db
nova/tests/functional/db/api
nova/tests/functional/compute
nova/tests/functional/notification_sample_tests
nova/tests/functional/api_sample_tests
nova/tests/functional/api_sample_tests/api_samples
nova/tests/functional/api_sample_tests/api_samples/os-networks
nova/tests/functional/api_sample_tests/api_samples/os-quota-sets-noop
nova/tests/functional/api_sample_tests/api_samples/images
nova/tests/functional/api_sample_tests/api_samples/os-evacuate
nova/tests/functional/api_sample_tests/api_samples/os-evacuate/v2.29
nova/tests/functional/api_sample_tests/api_samples/os-evacuate/v2.14
nova/tests/functional/api_sample_tests/api_samples/os-floating-ips
nova/tests/functional/api_sample_tests/api_samples/extension-info
nova/tests/functional/api_sample_tests/api_samples/os-networks-associate
nova/tests/functional/api_sample_tests/api_samples/server-ips
nova/tests/functional/api_sample_tests/api_samples/os-quota-sets
nova/tests/functional/api_sample_tests/api_samples/os-quota-sets/v2.36
nova/tests/functional/api_sample_tests/api_samples/os-quota-sets/v2.57
nova/tests/functional/api_sample_tests/api_samples/os-hide-server-addresses
nova/tests/functional/api_sample_tests/api_samples/os-hosts
nova/tests/functional/api_sample_tests/api_samples/os-certificates
nova/tests/functional/api_sample_tests/api_samples/os-volumes
nova/tests/functional/api_sample_tests/api_samples/os-volumes/v2.49
nova/tests/functional/api_sample_tests/api_samples/os-fping
nova/tests/functional/api_sample_tests/api_samples/os-pause-server
nova/tests/functional/api_sample_tests/api_samples/os-console-auth-tokens
nova/tests/functional/api_sample_tests/api_samples/os-preserve-ephemeral-rebuild
nova/tests/functional/api_sample_tests/api_samples/flavor-extra-specs
nova/tests/functional/api_sample_tests/api_samples/os-agents
nova/tests/functional/api_sample_tests/api_samples/os-shelve
nova/tests/functional/api_sample_tests/api_samples/os-floating-ips-bulk
nova/tests/functional/api_sample_tests/api_samples/servers
nova/tests/functional/api_sample_tests/api_samples/servers/v2.17
nova/tests/functional/api_sample_tests/api_samples/servers/v2.32
nova/tests/functional/api_sample_tests/api_samples/servers/v2.3
nova/tests/functional/api_sample_tests/api_samples/servers/v2.42
nova/tests/functional/api_sample_tests/api_samples/servers/v2.37
nova/tests/functional/api_sample_tests/api_samples/servers/v2.19
nova/tests/functional/api_sample_tests/api_samples/servers/v2.45
nova/tests/functional/api_sample_tests/api_samples/servers/v2.52
nova/tests/functional/api_sample_tests/api_samples/servers/v2.26
nova/tests/functional/api_sample_tests/api_samples/servers/v2.47
nova/tests/functional/api_sample_tests/api_samples/servers/v2.54
nova/tests/functional/api_sample_tests/api_samples/servers/v2.9
nova/tests/functional/api_sample_tests/api_samples/servers/v2.57
nova/tests/functional/api_sample_tests/api_samples/servers/v2.16
nova/tests/functional/api_sample_tests/api_samples/os-fixed-ips
nova/tests/functional/api_sample_tests/api_samples/os-fixed-ips/v2.4
nova/tests/functional/api_sample_tests/api_samples/os-floating-ip-dns
nova/tests/functional/api_sample_tests/api_samples/os-simple-tenant-usage
nova/tests/functional/api_sample_tests/api_samples/os-simple-tenant-usage/v2.40
nova/tests/functional/api_sample_tests/api_samples/os-create-backup
nova/tests/functional/api_sample_tests/api_samples/os-create-backup/v2.45
nova/tests/functional/api_sample_tests/api_samples/os-multiple-create
nova/tests/functional/api_sample_tests/api_samples/os-assisted-volume-snapshots
nova/tests/functional/api_sample_tests/api_samples/consoles
nova/tests/functional/api_sample_tests/api_samples/os-lock-server
nova/tests/functional/api_sample_tests/api_samples/os-baremetal-nodes
nova/tests/functional/api_sample_tests/api_samples/os-server-diagnostics
nova/tests/functional/api_sample_tests/api_samples/os-server-diagnostics/v2.48
nova/tests/functional/api_sample_tests/api_samples/os-remote-consoles
nova/tests/functional/api_sample_tests/api_samples/os-remote-consoles/v2.8
nova/tests/functional/api_sample_tests/api_samples/os-remote-consoles/v2.6
nova/tests/functional/api_sample_tests/api_samples/flavor-manage
nova/tests/functional/api_sample_tests/api_samples/flavor-manage/v2.55
nova/tests/functional/api_sample_tests/api_samples/os-suspend-server
nova/tests/functional/api_sample_tests/api_samples/os-security-groups
nova/tests/functional/api_sample_tests/api_samples/os-security-group-default-rules
nova/tests/functional/api_sample_tests/api_samples/servers-sort
nova/tests/functional/api_sample_tests/api_samples/os-hypervisors
nova/tests/functional/api_sample_tests/api_samples/os-hypervisors/v2.53
nova/tests/functional/api_sample_tests/api_samples/os-hypervisors/v2.33
nova/tests/functional/api_sample_tests/api_samples/os-hypervisors/v2.28
nova/tests/functional/api_sample_tests/api_samples/flavor-access
nova/tests/functional/api_sample_tests/api_samples/flavor-access/v2.7
nova/tests/functional/api_sample_tests/api_samples/os-server-groups
nova/tests/functional/api_sample_tests/api_samples/os-server-groups/v2.13
nova/tests/functional/api_sample_tests/api_samples/os-server-external-events
nova/tests/functional/api_sample_tests/api_samples/os-attach-interfaces
nova/tests/functional/api_sample_tests/api_samples/os-attach-interfaces/v2.49
nova/tests/functional/api_sample_tests/api_samples/versions
nova/tests/functional/api_sample_tests/api_samples/os-admin-password
nova/tests/functional/api_sample_tests/api_samples/server-metadata
nova/tests/functional/api_sample_tests/api_samples/keypairs
nova/tests/functional/api_sample_tests/api_samples/keypairs/v2.10
nova/tests/functional/api_sample_tests/api_samples/keypairs/v2.35
nova/tests/functional/api_sample_tests/api_samples/keypairs/v2.2
nova/tests/functional/api_sample_tests/api_samples/os-deferred-delete
nova/tests/functional/api_sample_tests/api_samples/os-cells
nova/tests/functional/api_sample_tests/api_samples/os-floating-ip-pools
nova/tests/functional/api_sample_tests/api_samples/os-instance-actions
nova/tests/functional/api_sample_tests/api_samples/os-instance-actions/v2.51
nova/tests/functional/api_sample_tests/api_samples/os-instance-actions/v2.21
nova/tests/functional/api_sample_tests/api_samples/os-instance-actions/v2.58
nova/tests/functional/api_sample_tests/api_samples/server-migrations
nova/tests/functional/api_sample_tests/api_samples/server-migrations/v2.59
nova/tests/functional/api_sample_tests/api_samples/server-migrations/v2.23
nova/tests/functional/api_sample_tests/api_samples/os-quota-class-sets
nova/tests/functional/api_sample_tests/api_samples/os-quota-class-sets/v2.57
nova/tests/functional/api_sample_tests/api_samples/os-quota-class-sets/v2.50
nova/tests/functional/api_sample_tests/api_samples/os-server-password
nova/tests/functional/api_sample_tests/api_samples/os-rescue
nova/tests/functional/api_sample_tests/api_samples/os-server-tags
nova/tests/functional/api_sample_tests/api_samples/os-server-tags/v2.26
nova/tests/functional/api_sample_tests/api_samples/limits
nova/tests/functional/api_sample_tests/api_samples/limits/v2.39
nova/tests/functional/api_sample_tests/api_samples/limits/v2.36
nova/tests/functional/api_sample_tests/api_samples/limits/v2.57
nova/tests/functional/api_sample_tests/api_samples/os-migrations
nova/tests/functional/api_sample_tests/api_samples/os-migrations/v2.59
nova/tests/functional/api_sample_tests/api_samples/os-migrations/v2.23
nova/tests/functional/api_sample_tests/api_samples/os-console-output
nova/tests/functional/api_sample_tests/api_samples/os-migrate-server
nova/tests/functional/api_sample_tests/api_samples/os-migrate-server/v2.25
nova/tests/functional/api_sample_tests/api_samples/os-migrate-server/v2.30
nova/tests/functional/api_sample_tests/api_samples/os-migrate-server/v2.56
nova/tests/functional/api_sample_tests/api_samples/os-services
nova/tests/functional/api_sample_tests/api_samples/os-services/v2.53
nova/tests/functional/api_sample_tests/api_samples/os-services/v2.11
nova/tests/functional/api_sample_tests/api_samples/os-availability-zone
nova/tests/functional/api_sample_tests/api_samples/os-aggregates
nova/tests/functional/api_sample_tests/api_samples/os-aggregates/v2.41
nova/tests/functional/api_sample_tests/api_samples/flavors
nova/tests/functional/api_sample_tests/api_samples/flavors/v2.55
nova/tests/functional/api_sample_tests/api_samples/os-multinic
nova/tests/functional/api_sample_tests/api_samples/os-tenant-networks
nova/tests/functional/api_sample_tests/api_samples/os-admin-actions
nova/tests/functional/api_sample_tests/api_samples/os-instance-usage-audit-log
nova/tests/live_migration
nova/tests/live_migration/hooks
nova/tests/unit
nova/tests/unit/api
nova/tests/unit/api/openstack
nova/tests/unit/api/openstack/compute
nova/tests/unit/api/openstack/placement
nova/tests/unit/network
nova/tests/unit/network/security_group
nova/tests/unit/keymgr
nova/tests/unit/db
nova/tests/unit/pci
nova/tests/unit/notifications
nova/tests/unit/notifications/objects
nova/tests/unit/compute
nova/tests/unit/compute/monitors
nova/tests/unit/compute/monitors/cpu
nova/tests/unit/objects
nova/tests/unit/privsep
nova/tests/unit/servicegroup
nova/tests/unit/image
nova/tests/unit/monkey_patch_example
nova/tests/unit/cells
nova/tests/unit/ssl_cert
nova/tests/unit/consoleauth
nova/tests/unit/volume
nova/tests/unit/fake_loadables
nova/tests/unit/cmd
nova/tests/unit/scheduler
nova/tests/unit/scheduler/client
nova/tests/unit/scheduler/filters
nova/tests/unit/scheduler/weights
nova/tests/unit/console
nova/tests/unit/console/rfb
nova/tests/unit/console/securityproxy
nova/tests/unit/api_samples_test_base
nova/tests/unit/conductor
nova/tests/unit/conductor/tasks
nova/console
nova/console/rfb
nova/console/securityproxy
nova/ipv6
nova/conf
nova/conductor
nova/conductor/tasks
nova/CA
nova/CA/projects
nova/CA/reqs
nova/CA/private
nova/CA/newcerts

Reply:
The packages installed at `/usr/local/lib/python2.7/dist-packages` should be available to all users of the system.

{code}
# unzip -l /home/vagrant/nova_lxd-17.0.1+1.0-py2.7.egg
Archive:  /home/vagrant/nova_lxd-17.0.1+1.0-py2.7.egg
  Length      Date    Time    Name
---------  ---------- -----   ----
     2767  2019-03-18 13:33   EGG-INFO/PKG-INFO
     2825  2019-03-18 13:33   EGG-INFO/SOURCES.txt
        1  2019-03-18 13:33   EGG-INFO/dependency_links.txt
       90  2019-03-18 13:33   EGG-INFO/entry_points.txt
        1  2019-03-12 17:27   EGG-INFO/not-zip-safe
       47  2019-03-18 13:33   EGG-INFO/pbr.json
      168  2019-03-18 13:33   EGG-INFO/requires.txt
       49  2019-03-18 13:33   EGG-INFO/top_level.txt
        0  2019-03-11 16:18   nova/tests/unit/virt/lxd/__init__.py
      159  2019-03-18 13:33   nova/tests/unit/virt/lxd/__init__.pyc
     9802  2019-03-11 16:18   nova/tests/unit/virt/lxd/fake_api.py
     9922  2019-03-18 13:33   nova/tests/unit/virt/lxd/fake_api.pyc
     3307  2019-03-11 16:43   nova/tests/unit/virt/lxd/stubs.py
     4179  2019-03-18 13:33   nova/tests/unit/virt/lxd/stubs.pyc
     2250  2019-03-11 16:18   nova/tests/unit/virt/lxd/test_common.py
     2651  2019-03-18 13:33   nova/tests/unit/virt/lxd/test_common.pyc
    55004  2019-03-11 16:43   nova/tests/unit/virt/lxd/test_driver.py
    48977  2019-03-18 13:33   nova/tests/unit/virt/lxd/test_driver.pyc
    16863  2019-03-11 16:43   nova/tests/unit/virt/lxd/test_flavor.py
    13974  2019-03-18 13:33   nova/tests/unit/virt/lxd/test_flavor.pyc
     3704  2019-03-11 16:18   nova/tests/unit/virt/lxd/test_migrate.py
     3527  2019-03-18 13:33   nova/tests/unit/virt/lxd/test_migrate.pyc
     3772  2019-03-11 16:18   nova/tests/unit/virt/lxd/test_session.py
     4030  2019-03-18 13:33   nova/tests/unit/virt/lxd/test_session.pyc
     9686  2019-03-11 16:18   nova/tests/unit/virt/lxd/test_storage.py
     7410  2019-03-18 13:33   nova/tests/unit/virt/lxd/test_storage.pyc
    12804  2019-03-11 16:43   nova/tests/unit/virt/lxd/test_vif.py
    13558  2019-03-18 13:33   nova/tests/unit/virt/lxd/test_vif.pyc
       63  2019-03-11 16:18   nova/virt/lxd/__init__.py
      233  2019-03-18 13:33   nova/virt/lxd/__init__.pyc
     1325  2019-03-11 16:18   nova/virt/lxd/common.py
      943  2019-03-18 13:33   nova/virt/lxd/common.pyc
    50337  2019-03-11 16:43   nova/virt/lxd/driver.py
    43969  2019-03-18 13:33   nova/virt/lxd/driver.pyc
     7812  2019-03-12 17:48   nova/virt/lxd/flavor.py
     7047  2019-03-18 13:33   nova/virt/lxd/flavor.pyc
     7664  2019-03-11 16:43   nova/virt/lxd/session.py
     6205  2019-03-18 13:33   nova/virt/lxd/session.pyc
     5872  2019-03-11 16:43   nova/virt/lxd/storage.py
     3476  2019-03-18 13:33   nova/virt/lxd/storage.pyc
     8458  2019-03-12 17:48   nova/virt/lxd/vif.py
     8574  2019-03-18 13:33   nova/virt/lxd/vif.pyc
      102  2019-03-11 16:18   nova_lxd_tempest_plugin/README
        0  2019-03-11 16:18   nova_lxd_tempest_plugin/__init__.py
      158  2019-03-18 13:33   nova_lxd_tempest_plugin/__init__.pyc
     1084  2019-03-11 16:18   nova_lxd_tempest_plugin/plugin.py
     1259  2019-03-18 13:33   nova_lxd_tempest_plugin/plugin.pyc
        0  2019-03-11 16:18   nova_lxd_tempest_plugin/tests/__init__.py
      164  2019-03-18 13:33   nova_lxd_tempest_plugin/tests/__init__.pyc
        0  2019-03-11 16:18   nova_lxd_tempest_plugin/tests/api/__init__.py
      168  2019-03-18 13:33   nova_lxd_tempest_plugin/tests/api/__init__.pyc
        0  2019-03-11 16:18   nova_lxd_tempest_plugin/tests/api/compute/__init__.py
      176  2019-03-18 13:33   nova_lxd_tempest_plugin/tests/api/compute/__init__.pyc
        0  2019-03-11 16:18   nova_lxd_tempest_plugin/tests/api/compute/servers/__init__.py
      184  2019-03-18 13:33   nova_lxd_tempest_plugin/tests/api/compute/servers/__init__.pyc
     6162  2019-03-11 16:43   nova_lxd_tempest_plugin/tests/api/compute/servers/test_create_server.py
     6528  2019-03-18 13:33   nova_lxd_tempest_plugin/tests/api/compute/servers/test_create_server.pyc
     4428  2019-03-11 16:18   nova_lxd_tempest_plugin/tests/api/compute/servers/test_servers.py
     4586  2019-03-18 13:33   nova_lxd_tempest_plugin/tests/api/compute/servers/test_servers.pyc
        0  2019-03-11 16:18   nova_lxd_tempest_plugin/tests/api/compute/volumes/__init__.py
      184  2019-03-18 13:33   nova_lxd_tempest_plugin/tests/api/compute/volumes/__init__.pyc
     4057  2019-03-11 16:43   nova_lxd_tempest_plugin/tests/api/compute/volumes/test_attach_volume.py
     4512  2019-03-18 13:33   nova_lxd_tempest_plugin/tests/api/compute/volumes/test_attach_volume.pyc
        0  2019-03-11 16:18   nova_lxd_tempest_plugin/tests/scenario/__init__.py
      173  2019-03-18 13:33   nova_lxd_tempest_plugin/tests/scenario/__init__.pyc
    30560  2019-03-11 16:43   nova_lxd_tempest_plugin/tests/scenario/manager.py
    22014  2019-03-18 13:33   nova_lxd_tempest_plugin/tests/scenario/manager.pyc
     5448  2019-03-11 16:43   nova_lxd_tempest_plugin/tests/scenario/test_server_basic_ops.py
     5486  2019-03-18 13:33   nova_lxd_tempest_plugin/tests/scenario/test_server_basic_ops.pyc
     4277  2019-03-11 16:43   nova_lxd_tempest_plugin/tests/scenario/test_volume_ops.py
     4400  2019-03-18 13:33   nova_lxd_tempest_plugin/tests/scenario/test_volume_ops.pyc
---------                     -------
   489615                     71 files
root@us-ral-5a-e2:/usr/local/lib/python2.7/dist-packages# tree nova
nova
├── tests
│   └── unit
│       └── virt
│           └── lxd
│               ├── __init__.py
│               ├── __init__.pyc
│               ├── fake_api.py
│               ├── fake_api.pyc
│               ├── stubs.py
│               ├── stubs.pyc
│               ├── test_common.py
│               ├── test_common.pyc
│               ├── test_driver.py
│               ├── test_driver.pyc
│               ├── test_flavor.py
│               ├── test_flavor.pyc
│               ├── test_migrate.py
│               ├── test_migrate.pyc
│               ├── test_session.py
│               ├── test_session.pyc
│               ├── test_storage.py
│               ├── test_storage.pyc
│               ├── test_vif.py
│               └── test_vif.pyc
└── virt
    └── lxd
        ├── __init__.py
        ├── __init__.pyc
        ├── common.py
        ├── common.pyc
        ├── driver.py
        ├── driver.pyc
        ├── flavor.py
        ├── flavor.pyc
        ├── session.py
        ├── session.pyc
        ├── storage.py
        ├── storage.pyc
        ├── vif.py
        └── vif.pyc

6 directories, 34 files
{code}

Reply:
At this point, I think we should go with the origin plan:

1. Include `nova/__init.py__` and `nova/virt/__init.py__` in wheel.
2. Use pip to install wheel to `/usr/local/lib/python2.7/dist-packages`